File: 프로그래머스/Lv.2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();

        for (int i = 0; i < clothes.length; i++) {
            if(!map.containsKey(clothes[i][1])){
                map.put(clothes[i][1], 1);
            }else{
                int value = map.get(clothes[i][1])+1;
                map.replace(clothes[i][1], value);
            }
        }
        int answer = 0;
        int value = 1;
        for (String s : map.keySet()) {
            value *= map.get(s)+1;
        }
        answer=value-1;
        return answer;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dict = {}
    res = 1
    
    for i in range(len(clothes)):
        value, key = clothes[i]
        if key in dict:
            dict[key] += 1
        else:
            dict[key] = 1
            
    for key, value in dict.items():
        res *= value+1
    return res-1